packages/core/src/entity/EntityAssigner.ts

```diff
@@ -43,6 +44,16 @@ export class EntityAssigner {
       }
 
       if ([ReferenceType.MANY_TO_ONE, ReferenceType.ONE_TO_ONE].includes(props[prop]?.reference) && Utils.isDefined(value, true) && EntityAssigner.validateEM(em)) {
+
+        // eslint-disable-next-line no-prototype-builtins
+        if (options.updateNestedEntities && entity.hasOwnProperty(prop) && (Utils.isEntity(entity[prop]) || Reference.isReference(entity[prop])) && Utils.isPlainObject(value)) {
+          const unwrappedEntity = Reference.isReference(entity[prop]) ? entity[prop].unwrap() : entity[prop];
```

Comment:
there is a helper for this already
```suggestion
          const unwrappedEntity = Reference.unwrapReference(entity[prop]);
```